### PR TITLE
Remove the hard dependency on PDO

### PR DESCRIPTION
**Pull request by fritzmg**
| Q                | A
| -----------------| ---
| Fixed issues     | Fixes #2459
| Docs PR or issue | -

Same as #2462, but for Contao 4.9+.

### CODE REVIEW

**Comment by fritzmg** [comment]
Not sure why I did not have to add the PHP `mysqli` extension to the `Windows` CI test in #2462 🤔

**Review by aschempp — DISMISSED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -188,7 +188,7 @@ jobs:
               uses: shivammathur/setup-php@v2
               with:
                   php-version: 7.3
-                  extensions: dom, fileinfo, filter, gd, hash, intl, json, mbstring, pcre, pdo_mysql, zlib
+                  extensions: dom, fileinfo, filter, gd, hash, intl, json, mbstring, mysqli, pcre, pdo_mysql, zlib
```
Comment (leofeyer) [inline .github/workflows/ci.yml:191]: Why is this necessary?

Comment (fritzmg) [reply .github/workflows/ci.yml:191]: The test fails on Windows without it. As I said, I am not sure why the test did not fail in #2462 without it too.

Comment (leofeyer) [reply .github/workflows/ci.yml:191]: I see.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @fritzmg.
